lib/openstack_query/query_blocks/query_builder.py

```diff
@@ -123,14 +141,70 @@ def parse_where(
             )
         else:
             logger.info(
-                "Found server-side filters for preset '%s': prop '%s' pair: {%s}",
+                "Found %s set of server-side filters for preset '%s': prop '%s' pair",
+                len(server_side_filters),
                 preset.name,
                 prop.name,
-                ", ".join(
-                    [f"{key}: '{val}'" for key, val in self.server_side_filters.items()]
-                ),
             )
 
+        logger.debug(
+            "server-side-filters found: %s",
+            {f"{server_side_filters}" if server_side_filters else "None"},
+        )
+        logger.debug(
+            "client-side-filter found: %s",
+            {client_side_filter if client_side_filter else "None"},
+        )
+
+        self._add_filter(
+            client_side_filter=client_side_filter,
+            server_side_filters=server_side_filters,
+        )
+
+    def _add_filter(
+        self,
+        client_side_filter: ClientSideFilterFunc,
+        server_side_filters: Optional[ServerSideFilters] = None,
+    ) -> None:
+        """
+        method which parses client-side and server-side filters for a given preset and adds it to the
+        list of query operations to perform
+        :param client_side_filter: A client side filter function for the query preset
+        :param server_side_filters: An optional set of server side filters for the query preset
+        """
+
+        # add as client_side_filter if no server_side_filter
+        if not server_side_filters:
+            self.client_side_filters.append(client_side_filter)
+            return
+
+        # we convert to singleton list for aggregating into server_side_filter
+        if not isinstance(server_side_filters, list):
+            server_side_filters = [server_side_filters]
+
+        # check each any new server filter being added has overlapping keys with any server filters currently added
+        # if so add as client_side_filter
+        for current_server_filter in self.server_side_filters:
+            for new_server_filter in server_side_filters:
+                if set(new_server_filter.keys()) & set(current_server_filter.keys()):
```

Comment:
Nit: use `and` instead of `&` 

Also your comments are straying into what it's doing rather than why, why do we append a client side filter if the new and old server side filter keys match?

Reply:
& does something else here - i changed to intersection